tests/test_modeling_common.py

```diff
@@ -3167,33 +3167,106 @@ def test_mismatched_shapes_have_properly_initialized_weights(self):
         configs_no_init = _config_zero_init(config)
 
         for model_class in self.all_model_classes:
-            if model_class.__name__ not in get_values(MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES):
+            mappings = [
+                MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_AUDIO_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING_NAMES,
+            ]
+            is_classication_model = any(model_class.__name__ in get_values(mapping) for mapping in mappings)
+
+            if not is_classication_model:
                 continue
```

Comment:
IMO, here is not return but just continue to the next `model_class`. If we skip here, it will exit the test method and the remaining model classes won't be tested.

Reply:
Ah, yes, you're right 🤦‍♀️